### PR TITLE
Karan react navigation

### PR DESCRIPTION
**Pull request by karanasthana**

### CODE REVIEW

File: habit/App.js [hunk 1]
```diff
@@ -117,12 +117,14 @@ import React, {Component} from 'react';
 import store from './src/store';
 import {Provider} from 'react-redux';
 import App from './src';
```
Comment (karanasthana) [inline habit/App.js:119]: remove

File: habit/src/AppScreen.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        // NOTE: This is to forcefully make the SplashScreen last for 2 minutes
+        setTimeout(function() {
+            self.setState({ isTimeDone: true });
+        }, 2000);
+    }
+
+    async checkCredentialsAndUpdate() {
+        // const credentials = await Keychain.getGenericPassword();
+        // if (!_.isEmpty(credentials.username)) {
+        if (false) {
```
Comment (karanasthana) [inline habit/src/AppScreen.js:28]: Add comment that this is to be used after backend API is functional

File: habit/src/containers/login/LoginScreen.js [hunk 3]
```diff
@@ -3,12 +3,12 @@ import { View } from 'react-native'
 import CustomTextInput from '../../components/TextInput'
 import * as CONSTANTS from '../../utils/string_constants'
 
-import styles from '../../styles/common'
```
Comment (karanasthana) [inline habit/src/containers/login/LoginScreen.js:6]: Revert

File: habit/src/AppNavigation.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+            // },
+            // headerTitleStyle: {
+            //     color: 'white',
+            // },
+            headerShown: false,
+        },
+    },
+);
+
+export const createRootSwitchNavigator = (signedOut = false, allProjectObjects = {}) => {
```
Comment (ayu-ayushisood) [inline habit/src/AppNavigation.js:54]: what is this object? allProjectObjects

File: habit/src/AppScreen.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    state = {
+        isLoading: true,
+        isSignedOut: true,
+        userToken: null,
+        isTimeDone: false,
+    };
+
+    componentDidMount() {
+        // eslint-disable-next-line consistent-this
+        let self = this;
```
Comment (ayu-ayushisood) [inline habit/src/AppScreen.js:17]: why have you assigned the value of this to self? I think **this** should work fine if you use arrow function in setTimeout

File: habit/src/AppScreen.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        this.checkCredentialsAndUpdate();
+        // NOTE: This is to forcefully make the SplashScreen last for 2 minutes
+        setTimeout(function() {
+            self.setState({ isTimeDone: true });
```
Comment (ayu-ayushisood) [inline habit/src/AppScreen.js:21]: use arrow function to prevent assigning value of **this** to some other variable

File: habit/src/AppScreen.js [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+    async checkCredentialsAndUpdate() {
+        // const credentials = await Keychain.getGenericPassword();
+        // if (!_.isEmpty(credentials.username)) {
+        // TODO Karan use this if-else to decide where to go, Login OR Home
+        if (false) {
+            const pwd = JSON.parse(credentials.password);
+            global.authorization = `ApiKey ${pwd.username}:${pwd.apiKey}`;
+            global.subdomain = pwd.accountSubdomain;
+            global.baseUrl = `${PROTOCOL}${pwd.accountSubdomain}${ORGZIT_URL_APP_FLAVOUR}`;
```
Comment (ayu-ayushisood) [inline habit/src/AppScreen.js:33]: what is orgzit doing here?

File: habit/src/AppScreen.js [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+        // eslint-disable-next-line consistent-this
+        let self = this;
+        this.checkCredentialsAndUpdate();
+        // NOTE: This is to forcefully make the SplashScreen last for 2 minutes
+        setTimeout(function() {
+            self.setState({ isTimeDone: true });
+        }, 2000);
+    }
+
+    async checkCredentialsAndUpdate() {
```
Comment (ayu-ayushisood) [inline habit/src/AppScreen.js:25]: why is this function async, when we are not using it as an asynchronous function?

File: habit/package.json [hunk 9]
```diff
@@ -10,9 +10,18 @@
     "lint": "eslint ."
   },
   "dependencies": {
+    "@react-native-community/masked-view": "^0.1.10",
```
Comment (ayu-ayushisood) [inline habit/package.json:13]: where are we using this library?

File: habit/ios/Podfile.lock [hunk 10]
```diff
@@ -236,6 +236,8 @@ PODS:
     - React-cxxreact (= 0.63.0)
     - React-jsi (= 0.63.0)
   - React-jsinspector (0.63.0)
+  - react-native-safe-area-context (3.1.1):
```
[the remark this answers is not in the document]
Comment (karanasthana) [reply habit/ios/Podfile.lock:239]: Already pushed in some other PR sometime, will add to .gitgnore in a separate PR

File: habit/src/AppNavigation.js [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+import { createStackNavigator } from 'react-navigation-stack';
+import { createSwitchNavigator } from 'react-navigation';
+
+let LoginStack = createStackNavigator(
+    {
+        Login: {
+            screen: Login,
+        },
+        // ForgotPassword: {
+        //     screen: ForgotPassword,
```
[the remark this answers is not in the document]
Comment (karanasthana) [reply habit/src/AppNavigation.js:12]: We need to uncomment the code as we integrate other screens

File: habit/src/AppScreen.js (hunk 7 above)
Comment (karanasthana) [reply habit/src/AppScreen.js:33]: Haha, removing

File: habit/src/AppScreen.js (hunk 8 above)
Comment (karanasthana) [reply habit/src/AppScreen.js:25]: This function will be async as soon as we implement the login API.
We will have to store the login credentials in a 3rd party library like Keychain (get from such libraries is async)

File: habit/package.json (hunk 9 above)
Comment (karanasthana) [reply habit/package.json:13]: React navigation is dependent on masked-view

Comment (ayu-ayushisood) [reply habit/package.json:13]: I don't think so

Comment (ayu-ayushisood) [reply habit/package.json:13]: I don't think so

**Pull request merged**